goagen/codegen/import_spec_test.go

```diff
@@ -0,0 +1,142 @@
+package codegen_test
+
+import (
+	. "github.com/goadesign/goa/design"
+	. "github.com/goadesign/goa/design/apidsl"
+	"github.com/goadesign/goa/dslengine"
+	"github.com/goadesign/goa/goagen/codegen"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("AttributeImports", func() {
```

Comment:
It would also be interesting to test the recursion in particular with cycles, something like:
```go
o := Object{}
child := &AttributeDefinition{Type: o}
po := Object{"child":child}
parent := &AttributeDefinition{Type:po}
o["parent"] = parent
```